sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -327,6 +326,32 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder {
         .getOrElse(Array.empty[String])
     val bucketSpec = Option(ctx.bucketSpec()).map(visitBucketSpec)
 
+    val path = new CaseInsensitiveMap(options).get("path")
+    val storage = CatalogStorageFormat(path, Some(provider), options)
+    val columns = Option(ctx.colTypeList()).toSeq.flatMap(visitCatalogColumns)
```

Comment:
Here, you might hit an issue here: the value of `columns` is `None`, but the partitioning/bucketing columns might be specified. Then, we will violate the following requirements set in class `CatalogTable`:

https://github.com/apache/spark/blob/42279bff686f9808ec7a9e8f4da95c717edc6026/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala#L139-L147

 Do you think we should move the checking to the other parts?